### PR TITLE
Fixes #3293 add FactoryGirl to tests

### PR DESCRIPTION
**Pull request by GregSutcliffe**
Also adds an example factory.

### CODE REVIEW

**Comment by domcleal** [comment]
Test failures:

<pre>
Installing factory_girl (4.2.0) 
Gem::InstallError: factory_girl requires Ruby version >= 1.9.2.
</pre>

I'd rather not have an unused example there, is there something (like settings) that's easy to convert?

**Comment by GregSutcliffe** [comment]
@domcleal pinned to Factorygirl 2.X for 1.8 compat. Added a real example involved migrated Usergroups - Settings is a bit of a special case as deleting the fixtures causing interesting effects with the initializers :P

**Comment by GregSutcliffe** [comment]
It should also be noted that "example" is a bit misleading - the code in factories/settings.rb is a valid Setting factory and is used by foreman_default_hostgroup to build a Setting without needing to define it's own Factory.

File: bundler.d/test.rb [hunk 1]
```diff
@@ -14,4 +14,5 @@
   gem 'database_cleaner', '0.9.1'
   gem 'launchy'
   gem 'spork'
+  gem 'factory_girl_rails', '~> 1.2', :require => false
```
Comment (domcleal) [inline bundler.d/test.rb:17]: 1.x or 2.x?

**Comment by domcleal** [comment]
You're going to hate this, but the integration tests failed :)

Noted about the example, I see, thanks.

File: bundler.d/test.rb (hunk 1 above)
Comment (GregSutcliffe) [inline bundler.d/test.rb:17]: This is correct, fg_rails 1 pulls in fg 2

Comment (GregSutcliffe) [inline bundler.d/test.rb:17]: https://github.com/thoughtbot/factory_girl/blob/v2.6.5/GETTING_STARTED.md#update-your-gemfile

Comment (domcleal) [inline bundler.d/test.rb:17]: Gotcha, thanks.

**Comment by dmitri-d** [comment]
More of a nitpick, but it's better to use sequences and generators instead of hard-coded values (you end up with better tests). https://github.com/thoughtbot/factory_girl/blob/master/GETTING_STARTED.md#lazy-attributes has a good example.

File: test/unit/usergroup_test.rb [hunk 2]
```diff
@@ -130,29 +132,29 @@ def setup_user operation
   end
 
   test "user with destroy permissions should be able to destroy" do
+    record = FactoryGirl.create(:usergroup)
     setup_user "destroy"
-    record =  Usergroup.first
```
Comment (dmitri-d) [inline test/unit/usergroup_test.rb:134]: Edit: NM!

**Comment by domcleal** [comment]
@witlessbird so random strings for things like names and descriptions?

**Comment by dmitri-d** [comment]
@domcleal: I usually try to automatically generate as many fields as possible, but yours are most typical examples.

**Comment by domcleal** [comment]
I like the idea, what do you think @GregSutcliffe?  The only place it matters is one integration test, where we could instead specify the name we care about in the create call.  (Just thinking to get us into "best practices" from the start.)

Otherwise looking good I think.

**Comment by GregSutcliffe** [comment]
Trivial to add if you want it as a first example.

**Comment by domcleal** [comment]
I think it'd be good if it gets us using it well, as people will only start copying your first examples :)

**Comment by GregSutcliffe** [comment]
Copying from me is generally a bad idea.... :) I'll get that added now.

**Comment by GregSutcliffe** [comment]
Sadly `generate(:random_string)` appears to be a FG4 thing, but I've added some sequences and converted some of the basic Usergroup model tests to call FG.build where appropriate for in-memory tests. Looks nice actually :)

@domcleal @witlessbird how does that look to you for an example now?

**Comment by dmitri-d** [comment]
@GregSutcliffe: looks good! (you may know this, but you can re-use sequences in multiple factories, but only if you define them outside of "factory" scope. This mostly compensates lacks of 'generate').

**Comment by GregSutcliffe** [comment]
Ah yes, I knew you could re-use them, but hadn't made the jump to a generate replacement. Maybe later, I cba altering it again :)

**Comment by domcleal** [comment]
Thanks @GregSutcliffe, merged as 3035495 for Foreman 1.3.1 as requested.

**Pull request closed without merging**